lua/lazyvim/plugins/extras/lang/yaml.lua

```diff
@@ -74,4 +74,41 @@ return {
       },
     },
   },
+
+  -- support auto-detecting and selecting yaml schemas.
+  -- use ":Telescope yaml_schema" to change current schema.
+  {
+    "someone-stole-my-name/yaml-companion.nvim",
+    ft = { "yaml" },
+    dependencies = {
+      { "neovim/nvim-lspconfig" },
+      { "nvim-lua/plenary.nvim" },
+      { "nvim-telescope/telescope.nvim" },
+    },
+    config = function(_, opts)
+      local cfg = require("yaml-companion").setup(opts)
+      require("lspconfig")["yamlls"].setup(cfg)
+      require("telescope").load_extension("yaml_schema")
+    end,
+  },
+
+  -- Show currently used yaml-schema on lualine status bar
+  {
```

Comment:
I have some thoughts on this though. Instead of including it on default, maybe we should add it as optional.  Some users have their own lua line config like me and this will break things on them or make it look weird

Reply:
That's not what `optional` does. `optional` means that it will only be set up if there's another instance of the same plugin somewhere in another fragment of the plugin spec. 

I agree that the lualine spec in this PR should be `optional` however, as there might be other users that use a different statusline and it would be better to decouple the use of this PR from Lualine as a hard dependency with it.

Reply:
Yes, that's the same. You would do something like `{ "lualine.nvim", optional = true, ...}` here as well. Or am I misunderstanding what you're saying?

Reply:
Yeah. I was just talking about removing the lualine from the PR and adding a section on the docs similar to [lazyvim.org/extras/coding/copilot#lualinenvim-optional](https://www.lazyvim.org/extras/coding/copilot#lualinenvim-optional)

Reply:
There's no need to remove it. Just guard it with `optional` like all the rest of the extras do to take into consideration if the plugin in name is actually enabled or not.

And if you're worried about this PR messing with your lualine that's unavoidable. The only thing you can do (as far as I know) is remove in your personal configuration the lualine components you don't want to see. That's what I'm doing as well.

Reply:
Yeah I am thinking thats unavoidable. Will have to manually remove the component from lualine after this PR gets merged, atleast thats also the easiet solution I can think of atm